### PR TITLE
[Merged by Bors] - chore: adaptations for nightly-2024-09-30

### PR DESCRIPTION
**Pull request by jcommelin**

### CODE REVIEW

File: Mathlib/Data/List/Basic.lean [hunk 1]
```diff
@@ -1331,13 +1331,6 @@ local notation a " ⋆ " b => op a b
 /-- Notation for `foldl op a l`. -/
 local notation l " <*> " a => foldl op a l
 
-theorem foldl_assoc : ∀ {l : List α} {a₁ a₂}, (l <*> a₁ ⋆ a₂) = a₁ ⋆ l <*> a₂
```
Comment (j-loreaux) [inline Mathlib/Data/List/Basic.lean:1334]: I guess this was upstreamed? If so, just resolve this convo.

File: Mathlib/Data/Nat/Defs.lean [hunk 2]
```diff
@@ -137,7 +137,12 @@ lemma one_lt_iff_ne_zero_and_ne_one : ∀ {n : ℕ}, 1 < n ↔ n ≠ 0 ∧ n ≠
 
 lemma le_one_iff_eq_zero_or_eq_one : ∀ {n : ℕ}, n ≤ 1 ↔ n = 0 ∨ n = 1 := by simp [le_succ_iff]
 
-@[simp] lemma lt_one_iff : n < 1 ↔ n = 0 := Nat.lt_succ_iff.trans <| by rw [le_zero_eq]
+#adaptation_note
+/-- This was accidently left off when upstreamed in Lean in
+https://github.com/leanprover/lean4/pull/5478.
+It can be removed again after nightly-2024-09-28.
+-/
+attribute [simp] lt_one_iff
```
Comment (j-loreaux) [inline Mathlib/Data/Nat/Defs.lean:145]: Given that this is adapations for nightly 2024-09-30, can't we remove this now?

File: Mathlib/Data/Nat/Defs.lean [hunk 3]
```diff
@@ -1011,9 +1016,6 @@ lemma div_ne_zero_iff_of_dvd (hba : b ∣ a) : a / b ≠ 0 ↔ a ≠ 0 ∧ b ≠
 @[simp] lemma mul_mod_mod (a b c : ℕ) : (a * (b % c)) % c = a * b % c := by
   rw [mul_mod, mod_mod, ← mul_mod]
 
-@[simp] lemma mod_mul_mod (a b c : ℕ) : (a % c * b) % c = a * b % c := by
```
Comment (j-loreaux) [inline Mathlib/Data/Nat/Defs.lean:1014]: why are we removing this? was it upstreamed?

File: Mathlib/Data/Option/Defs.lean [hunk 4]
```diff
@@ -75,10 +75,6 @@ abbrev iget [Inhabited α] : Option α → α
 theorem iget_some [Inhabited α] {a : α} : (some a).iget = a :=
   rfl
 
-@[simp]
-theorem mem_toList {a : α} {o : Option α} : a ∈ toList o ↔ a ∈ o := by
```
Comment (j-loreaux) [inline Mathlib/Data/Option/Defs.lean:79]: ditto

**Review by j-loreaux — COMMENTED** [review]
bors d+

File: Mathlib/AlgebraicGeometry/Morphisms/QuasiCompact.lean [hunk 5]
```diff
@@ -198,7 +198,15 @@ theorem exists_pow_mul_eq_zero_of_res_basicOpen_eq_zero_of_isAffineOpen (X : Sch
     {U : X.Opens} (hU : IsAffineOpen U) (x f : Γ(X, U))
     (H : x |_ X.basicOpen f = 0) : ∃ n : ℕ, f ^ n * x = 0 := by
   rw [← map_zero (X.presheaf.map (homOfLE <| X.basicOpen_le f : X.basicOpen f ⟶ U).op)] at H
-  obtain ⟨n, e⟩ := (hU.isLocalization_basicOpen f).exists_of_eq H
+  #adaptation_note
+  /--
+  Prior to nightly-2024-09-29, we could use dot notation here:
```
Comment (eric-wieser) [inline Mathlib/AlgebraicGeometry/Morphisms/QuasiCompact.lean:203]: Should this be an `adaptation_note`?

File: Mathlib/Data/Nat/Defs.lean (hunk 3 above)
Comment (kim-em) [reply Mathlib/Data/Nat/Defs.lean:1014]: Upstreamed, yes.

File: Mathlib/Data/Option/Defs.lean (hunk 4 above)
Comment (kim-em) [reply Mathlib/Data/Option/Defs.lean:79]: ditto!

File: Mathlib/Data/Nat/Defs.lean (hunk 2 above)
Comment (kim-em) [reply Mathlib/Data/Nat/Defs.lean:145]: Good catch. (There was a delay doing this adaptation because of an unfixable failure until this nightly arrived.)

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**